cfme/tests/containers/test_pause_resume_workers.py

```diff
@@ -0,0 +1,69 @@
+from cfme.containers.provider import ContainersProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
+import pytest
+
+
+pytestmark = [
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+
+def get_pause_resume_buttons(view):
+    resume_option = [menu_item for menu_item in view.toolbar.configuration.items if
+                     "resume this containers provider" in menu_item.lower()]
+    # ensure resume option is exists
+    assert resume_option, "No resume provider option is available in the configuration menu"
+    resume_option = resume_option.pop()
+
+    pause_option = [menu_item for menu_item in view.toolbar.configuration.items if
+                    "pause this containers provider" in menu_item.lower()]
+    # ensure pause option is exists
+    assert pause_option, "No pause provider option is available in the configuration menu"
+    pause_option = pause_option.pop()
+
+    return pause_option, resume_option
+
+
+def check_ems_state_in_diagnostics(appliance, provider):
+    workers_view = navigate_to(appliance.collections.diagnostic_workers, 'AllDiagnosticWorkers')
+    workers_view.browser.refresh()
+    try:
+        rows = workers_view.workers_table.rows(
+            name='Event Monitor for Provider: {}'.format(provider.name))
+        for r in rows:
```

Comment:
This is a bit of an odd pattern - from the looks of it rows yields or returns `None` when no rows match. Thoughts on a pattern like:

```
try:
    rows = table.rows(name='filter')
except Exception:
    return False
else:
    return rows is not None
```

Reply:
rows is always not None , since table.rows(name='filter') returns a generator and not a list.
I have to check that the generator does not return any elements.
I tried your suggestion while writing the code, and it didn't work, it will always return true.

Reply:
How about - 
        try:
            for r in workers_view.workers_table.rows(
                 name='Event Monitor for Provider: {}'.format(provider.name))
                 return True
        except Exception:
                return False

Reply:
Thanks for confirming @tkhamis, you're very right, its always a generator. I think we should take advantage of that fact, and using @sshveta example get a little more explicit with the exceptions we're handling here.

Little snippet from ipython:
```
# v is view of instances table (doesn't matter for example)
In [19]: v.entities.elements.rows().next().read()
Out[19]: 
{0: u'',
 1: u'',
 u'Allocated Size': u'170 GB',
 u'Architecture': u'64 bit',
 u'Availability Zone': <redacted>,
 u'Compliant': u'',
 u'Flavor': u'm1.small',
 u'IP Addresses': <redacted>,
 u'Last Analysis Time': u'',
 u'Load Balancer Status': u'',
 u'Name': u'cu-24x7',
 u'Provider': <redacted>,
 u'Region': u'Region 0'}

# Example of explicit pattern, using next() and StopIteration exception
In [20]: def check(name_match):
    ...:     try:
    ...:         v.entities.elements.rows(name=name_match).next()
    ...:         return True
    ...:     except StopIteration:
    ...:         # no matching rows
    ...:         return False
    ...:     

In [21]: check('cu-24x7')
Out[21]: True

In [22]: check('junk-name-fake')
Out[22]: False
```